src/Compilers/Core/Desktop/CommandLine/CommonCompiler.cs

```diff
@@ -258,17 +254,35 @@ internal bool PrintErrors(IEnumerable<DiagnosticInfo> diagnostics, TextWriter co
             return hasErrors;
         }
 
-        internal virtual void PrintError(DiagnosticInfo diagnostic, TextWriter consoleOutput)
+        protected virtual void PrintError(DiagnosticInfo diagnostic, TextWriter consoleOutput)
         {
             consoleOutput.WriteLine(diagnostic.ToString(Culture));
         }
 
+        private bool TryGetErrorLogger(TextWriter consoleOutput, out ErrorLogger errorLogger, CancellationToken cancellationToken)
+        {
+            errorLogger = null;
+            if (Arguments.ErrorLogPath != null)
+            {
+                var errorLog = OpenFile(Arguments.ErrorLogPath, consoleOutput, FileMode.Create, FileAccess.Write, FileShare.ReadWrite | FileShare.Delete);
+                if (errorLog == null)
+                {
+                    return false;
+                }
+
+                errorLogger = new ErrorLogger(errorLog, GetToolName(), GetAssemblyFileVersion(), GetAssemblyVersion());
+            }
+
+            return true;
```

Comment:
It's a little odd that TryGet will return true even when there is no logger. Maybe the check for errorlogpath should probably be split out from this method?